core/src/main/java/io/substrait/relation/ProtoRelConverter.java

```diff
@@ -361,12 +361,7 @@ protected VirtualTableScan newVirtualTable(ReadRel rel) {
 
   protected Fetch newFetch(FetchRel rel) {
     var input = from(rel.getInput());
-    var builder = Fetch.builder().input(input).offset(rel.getOffset());
-    if (rel.getCount() != -1) {
-      // -1 is used as a sentinel value to signal LIMIT ALL
-      // count only needs to be set when it is not -1
-      builder.count(rel.getCount());
-    }
+    var builder = Fetch.builder().input(input).offset(rel.getOffset()).count(rel.getCount());
```

Comment:
while the idea of not setting count if it's -1 is fine, this makes roundtrip tests fail if count is set in the pojo. Alternative fix is to ensure in the pojo it's never set if -1.